lib/include/tick/array/sparsearray2d.h

```diff
@@ -135,6 +185,62 @@ Array2d<T> BaseArray2d<T>::as_array2d() {
     return c;
 }
 
+// Using save/load methods directory on SparseArray2d<T> does not compile simply
+namespace cereal
+{
+  template <typename T, class Archive>
+  void save(Archive& ar, const SparseArray2d<T>& s) {
+    try{
+      ar(s._size_sparse);
+      ar(s._n_rows);
+      ar(s._n_cols);
+      ar(s._size);
+      ar( cereal::binary_data(s._data, sizeof(T) * s._size_sparse));
+      ar( cereal::binary_data(s._indices, sizeof(INDICE_TYPE) * s._size_sparse));
+      ar( cereal::binary_data(s._row_indices, sizeof(INDICE_TYPE) * (s._n_rows + 1)));
+    }catch(const std::exception& e){
+      std::cerr << e.what() << std::endl;
+    }
+  }
+
+  template <typename T, class Archive>
+  void load(Archive& ar, SparseArray2d<T>& s) {
+
+    if(s._data || s._indices || s._row_indices) 
+      throw std::runtime_error(
+        "SparseArray2d being used for deserializing may not have previous allocations");
+
+    try{
+      ar(s._size_sparse);
+      ar(s._n_rows);
+      ar(s._n_cols);
+      ar(s._size);
+
+      // using data structures directly on SparseArray2d<T> causes segfaults 
+      //  when deserializing, but using intermediary is ok, and 
+      //  doesn't (seem to) cause memory leaks
+      auto* s_data = new T[s._size_sparse];
+      auto* s_indices = new INDICE_TYPE[s._size_sparse];
+      auto* s_row_indices = new INDICE_TYPE[s._n_rows + 1];
+
+      ar( cereal::binary_data(s_data, sizeof(T) * s._size_sparse));
+      ar( cereal::binary_data(s_indices, sizeof(INDICE_TYPE) * s._size_sparse));
+      ar( cereal::binary_data(s_row_indices, sizeof(INDICE_TYPE) * (s._n_rows + 1)));
+
+      s._data        = s_data;
+      s._indices     = s_indices;
+      s._row_indices = s_row_indices;
+
+      s.is_data_allocation_owned = 1;
+      s.is_indices_allocation_owned = 1;
+      s.is_row_indices_allocation_owned = 1;
+
+    }catch(const std::exception& e){
+      std::cerr << e.what() << std::endl;
```

Comment:
maybe should we throw an exception? For example with TICK_ERROR macro

Reply:
sure